runtime/bcverify/staticverify.c

```diff
@@ -1893,8 +1939,16 @@ j9bcv_verifyClassStructure (J9PortLibrary * portLib, J9CfrClassFile * classfile,
 		}
 		if (isInit) {
 			BOOLEAN invalidRetType = FALSE;
-			if ((CFR_METHOD_NAME_INIT == isInit) /* Check return type of <init> on VTs, it will eventually be changed to <new>. */
-				&& J9_IS_CLASSFILE_VALUETYPE(classfile)
+			if (
+#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
+#ifdef J9VM_OPT_VALHALLA_NEW_FACTORY_METHOD
+				(CFR_METHOD_NAME_NEW == isInit)
```

Comment:
I thought we always wanted to run this code if the method name was `<new>`, and the only reason we were checking the type of the class file was because it was the only way to differentiate between a regular `<init>` and an `<init>` that behaved like a `<new>`. But I can change it to also check `J9_IS_CLASSFILE_VALUETYPE()` if you think that makes more sense.

Reply:
OK, the current change looks fine. You can add `to differentiate between a regular <init> and an <init> that behaved like a <new>` as a comment into the code.

Reply:
added in 480edf8